dashboard/test/controllers/api/v1/sections_students_controller_test.rb

```diff
@@ -81,6 +81,44 @@ class Api::V1::SectionsStudentsControllerTest < ActionController::TestCase
     assert @student.valid_password?('testpassword')
   end
 
+  test 'teacher can reset secret picture and words for their student' do
+    sign_in @teacher
+    @student.reload
+    old_secret_picture_path = @student.secret_picture.path
+    old_secret_words = @student.secret_words
+    params = {section_id: @section.id, id: @student.id, student: {name: 'testname'}, secrets: User::RESET_SECRETS}
+
+    put :update, params: params
```

Comment:
The `update` method is doc'd as `PATCH /sections/<section_id>/students/<id>` - why do we `put :update` here? (Not sure it's wrong, genuinely curious.)

Reply:
good point. from my understanding, rails basically mapped `put` to `patch` ([here's](https://github.com/rails/rails/issues/348) a PR to rails from 2011), but i just found a good explanation on the [rails blog](https://weblog.rubyonrails.org/2012/2/26/edge-rails-patch-is-the-new-primary-http-method-for-updates/) about when to use `put` versus `patch`.

based on that, i would think in basically all of the tests for this endpoint we should change `put` to `patch`